src/components/customers/AddCustomerDrawer.tsx

```diff
@@ -67,14 +82,73 @@ export const AddCustomerDrawer = forwardRef<DrawerRef, AddCustomerDrawerProps>(
           syncWithProvider: customer?.providerCustomer?.syncWithProvider ?? false,
         },
         paymentProvider: customer?.paymentProvider ?? undefined,
+        metadata: customer?.metadata ?? undefined,
       },
       validationSchema: object().shape({
         name: string().required(''),
         externalId: string().required(''),
+        metadata: array().of(
+          object().shape({
+            key: string().test({
+              test: function (value, { createError, path }) {
+                if (!value) {
+                  return false
+                }
+
+                // Check key unicity
+                // Compare key index and last index values
+                if (
+                  arguments[1].from[1]?.value?.metadata
+                    ?.map((m: { key: string }) => m.key)
+                    ?.indexOf(value) !==
+                  arguments[1].from[1]?.value?.metadata
+                    ?.map((m: { key: string }) => m.key)
+                    ?.lastIndexOf(value)
+                ) {
+                  return createError({
+                    path,
+                    message: MetadataErrorsEnum.uniqueness,
+                  })
+                }
+
+                if (value.length > 20) {
+                  return createError({
+                    path,
+                    message: MetadataErrorsEnum.maxLength,
+                  })
+                }
+
+                return true
+              },
+            }),
+            value: string().test({
+              test: (value, { createError, path }) => {
+                if (!value) return false
+                if (value.length > 40) {
+                  return createError({
+                    path,
+                    message: MetadataErrorsEnum.maxLength,
+                  })
+                }
+
+                return true
+              },
+            }),
+          })
+        ),
       }),
       validateOnMount: true,
       enableReinitialize: true,
       onSubmit: async (values, formikBag) => {
+        values?.metadata?.forEach(
+          (
+            element: CustomerMetadataInput & {
+              localId?: string
+            }
+          ) => {
+            delete element.localId
+          }
+        )
         const answer = await onSave(values)
```

Comment:
By doing this you modify the formikProps.values datas without "informing" which can lead to side effects (especially if the mutation does not work, your list of elements might be f****ed)
To avoid that, you could just : 
```tsx
const answer = await onSave(values?.metadata?.map(({ lodalId, ...metadata }) => metadata))
```
That way you don't modify the initial object